File: R/barcharts.r
## Load commodities data
data <- read.csv(file="../data/commodityentries.csv",head=TRUE,sep=",")
colnames(data)  <- c("Year","Butter","Cheese","Potatoes (maincrop)","Potatoes (earlies)")#Fix for text importing
library(reshape2)
data <- melt(data,"Year",variable.name="Commodity",value.name="Number")

## Load library
library(ggplot2)

## Grouped Bar Chart
# Initialise plot with x,y and grouping (fill) defines
ggplot(data=data, aes(x=Year,y=Number,fill=Commodity)) +
# Plot bars using position_dodge to group rather than stack
geom_bar(stat="identity", position=position_dodge()) +
# Legend is added automatically
# Change colours & legend labels
scale_fill_brewer(palette = "Greens",labels=c("Spreadable","Chunkable","Mashable","Boilable")) +
# Add axis labels
xlab("Years") +
ylab("Number of Data Entries")

## Stacked Bar Chart
# Initialise plot with x,y and grouping (fill) defines
ggplot(data=data, aes(x=Year,y=Number,fill=Commodity)) +
# Plot stacked bars
geom_bar(stat="identity") +
# Change axis label font size
theme(axis.title=element_text(size=20)) +
# Add axis labels
xlab("Years") +
ylab("Number of Data Entries")

## Faceted Bar Chart
# Initialise plot with x,y and grouping (fill) defines
ggplot(data=data, aes(x=Year,y=Number,fill=Commodity)) +
# Plot bars
geom_bar(stat="identity") +
# Make faceted
facet_wrap(~Commodity) +
# Add axis labels of specific size
xlab("Years") +
ylab("Number of Data Entries") +
# Change the overall 'theme'
theme_grey()

# Horizontal Bar Chart
data <- read.csv(file="../data/commodityentries.csv",head=TRUE,sep=",")
# Initialise plot with x,y
ggplot(data=data, aes(x=Year,y=Butter)) +
# Plot bars
geom_bar(stat="identity") +
# Rotate coordinate system
coord_flip() +
# Legend is added automatically
# Add axis labels
xlab("Years") +
ylab("Number of Data Entries")

File: MATLAB/barcharts.m
%% Load commodities data
filename = '../data/commodityentries.csv';
data = csvread(filename,1);
headings = {'Year','Butter','Cheese','Potatoes (main)', 'Potatoes (early)'};

%% Grouped Bar Chart
figure, bar(data(:,1),data(:,2:end));
set(gca,'FontSize',16)
% Add legend (outside plot area)
legend({headings{2},headings{3},headings{4},headings{5}},'FontSize',16,'Location','bestoutside','Orientation','horizontal')
% Change colours
colormap('jet')
% Add axis labels
xlabel('Years','FontSize',16)
ylabel('Number of Data Entries','FontSize',16)

%% Stacked Bar Chart
figure, bar(data(:,1),data(:,2:end),'stacked')
set(gca,'FontSize',20);%Set a font size for axis tick labels, note gca means 'get current axis;
% Add legend (inside plot area)
legend(headings{2},headings{3},headings{4},headings{5},'Location','best')
% Add axis labels of specific size
xlabel('Years','FontSize',20)
ylabel('Number of Data Entries','FontSize',20)

%% Bar Chart with Highlight
figure, b=bar(data(:,1),data(:,2:end));
% Highlight bar
b(2).EdgeColor='red';

%% Horizontal Bar Chart
figure, barh(data(:,1),data(:,2),'stacked')
% Add axis labels of specific size
xlabel('Years','FontSize',10)
ylabel('Number of Data Entries','FontSize',10)